samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.redwood.reposearch.android.views
+
+import com.example.redwood.reposearch.treehouse.HostApi
+import kotlinx.coroutines.suspendCancellableCoroutine
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okio.IOException
+
+class RealHostApi(
+  private val client: OkHttpClient,
+) : HostApi {
+  override suspend fun httpCall(url: String, headers: Map<String, String>): String {
+    return suspendCancellableCoroutine { continuation ->
+      val call = client.newCall(
+        Request.Builder()
+          .url(url)
+          .headers(headers.toHeaders())
+          .build(),
+      )
+      call.enqueue(
+        object : Callback {
+          override fun onFailure(call: Call, e: IOException) {
+            continuation.resumeWith(Result.failure(e))
+          }
+
+          override fun onResponse(call: Call, response: Response) {
+            val responseString = response.body!!.string()
```

Comment:
You need to check if `response.isSuccess` and otherwise resume with some kind of HTTP exception for non-2xx codes before reading the body. Otherwise you could be reading the body of a 404 or 500 or something.